src/api/firebase.js

```diff
@@ -77,6 +82,71 @@ export async function updateItem(listId, itemId, nextData) {
 	return await updateDoc(itemRef, nextData);
 }
 
+//create an urgency flag for sorting the list
+const urgencyFlag = (daysBetween) => {
+	if (daysBetween >= 30) {
+		return 'Not Soon';
+	} else if (daysBetween >= 7) {
+		return 'Kind of Soon';
+	} else if (daysBetween >= 0) {
+		return 'Soon';
+	} else if (daysBetween <= -1) {
+		return 'Overdue';
+	}
+	return '';
+};
+
+//assign the appropriate urgency flag to each item
+export async function comparePurchaseUrgency(itemList) {
+	itemList.forEach((item) => {
```

Comment:
I'm a little surprised calling `forEach` on this array and adding the new `urgency` property to each object works. It does work so that's cool, but, in the future, I would recommend using the `map` array method to return a new array as a more idiomatic modern Javascript approach.

You can chain array methods too using this approach, for rough example:

```
return itemList
    .map(() => ... )
    .sort(() => ... )
```